### PR TITLE
feat(tailscale)!: Update tailscale to Oauth

### PR DESCRIPTION
**Pull request by yevgenypats**
Update tailscale to use recommended (new) OAuth authentication.

Also, make few minor improvements, doc updates and make it ready for a handoff. Blogpost will come up later.

### CODE REVIEW

File: website/pages/docs/plugins/sources/tailscale/overview.mdx [hunk 1]
```diff
@@ -14,5 +14,50 @@ resources and loads it into any supported CloudQuery destination (e.g. PostgreSQ
 ## Authentication
 
 In order to fetch information from Tailscale, `cloudquery` needs to be authenticated.
-An API key is required for authentication.
-Get your API key from [Tailscale Keys Settings Page](https://login.tailscale.com/admin/settings/keys).
+The suggested way is to use an [OAuth client](https://tailscale.com/kb/1215/oauth-clients/), the previous way of
+using An [API key](https://login.tailscale.com/admin/settings/keys) is also supported.
```
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/tailscale/overview.mdx:18]:
```suggestion
The suggested way is to use an [OAuth client](https://tailscale.com/kb/1215/oauth-clients/). The previous way of
using an [API key](https://login.tailscale.com/admin/settings/keys) is also supported.
```

File: plugins/source/tailscale/resources/plugin/tables.go [hunk 2]
```diff
@@ -14,7 +14,9 @@ func tables() []*schema.Table {
 	return []*schema.Table{
 		acl.Acls(),
 		device.Devices(),
-		dns.Dns(),
+		dns.Nameservers(),
```
Comment (erezrokah) [inline plugins/source/tailscale/resources/plugin/tables.go:17]: Worth removing `// Code generated by codegen; DO NOT EDIT.` from the file too I think

File: website/pages/docs/plugins/sources/tailscale/overview.mdx [hunk 3]
```diff
@@ -14,5 +14,50 @@ resources and loads it into any supported CloudQuery destination (e.g. PostgreSQ
...
+- `api_key` (string, optional):
+  This way of authentication is not recommended. An API key to access Tailscale resources.
+  This can be obtained from [Tailscale Keys Settings Page](https://login.tailscale.com/admin/settings/keys).
+
+- `endpoint_url` (string, optional. Default: not used):
+  Endpoint URL to make the API requests to.
+
+## Example
+
+This example syncs from Tailscale to a Postgres destination, using `api_key` authentication.
```
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/tailscale/overview.mdx:44]:
```suggestion
This example syncs from Tailscale to a Postgres destination, using OAuth authentication.
```

File: website/pages/docs/plugins/sources/tailscale/overview.mdx [hunk 4]
```diff
@@ -14,5 +14,50 @@ resources and loads it into any supported CloudQuery destination (e.g. PostgreSQ
...
+spec:
+  name:    tailscale
+  path:    cloudquery/tailscale
+  version: "VERSION_SOURCE_TAILSCALE"
+  tables: [ "*" ]
+  destinations: [ "postgresql" ]
+
+  # plugin specific configuration
+  spec:
+    client_id:    "<YOUR_API_KEY_HERE>"
```
Comment (hermanschaaf) [inline website/pages/docs/plugins/sources/tailscale/overview.mdx:59]:
```suggestion
    client_id:    "<YOUR_CLIENT_ID_HERE>"
```

File: website/pages/docs/plugins/sources/tailscale/configuration.md [hunk 5]
```diff
@@ -1,37 +1,3 @@
 # Tailscale Source Plugin Configuration Reference
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/tailscale/configuration.md:1]: Should we remove this file? We can add a redirect from it to `overview`

File: plugins/source/tailscale/client/client.go [hunk 6]
```diff
@@ -12,35 +16,69 @@ import (
...
+			"client_secret": {pluginSpec.ClientSecret},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("error getting keys from %s: %w", oatuhURL, err)
+		}
+	
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, fmt.Errorf("error reading response body from %s: %w", oatuhURL, err)
+		}
```
Comment (hermanschaaf) [inline plugins/source/tailscale/client/client.go:59]:
```suggestion
		body, err := io.ReadAll(resp.Body)
		if err != nil {
			return nil, fmt.Errorf("error reading response body from %s: %w", oatuhURL, err)
		}
		defer body.Close()
```

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
LGTM, just a couple of doc updates and linting errors

File: plugins/source/tailscale/client/testing.go [hunk 7]
```diff
@@ -23,31 +25,26 @@ func MockTestHelper(t *testing.T, table *schema.Table, createService func(*httpr
...
+		t.Fatalf("failed to createService: %v", err)
+	}
+	var acl tailscale.ACL
+	if err := faker.FakeObject(&acl); err != nil {
+		t.Fatalf("failed to fake ACL: %v", err)
+	}
 
-	newTestExecutionClient := func(_ context.Context, logger zerolog.Logger, _ specs.Source, _ source.Options) (schema.ClientMeta, error) {
-		err := createService(mux)
+	mux.GET("/api/v2/tailnet/:tailnet/acl", func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
```
Comment (erezrokah) [inline plugins/source/tailscale/client/testing.go:36]: Should this code be under `createAcls` in `plugins/source/tailscale/resources/services/acl/acls_mock_test.go`?

**Review by erezrokah — CHANGES_REQUESTED** [review]
Added a few comments, also there are a bunch of lint errors to fix

File: plugins/source/tailscale/client/testing.go (hunk 7 above)
Comment (yevgenypats) [reply plugins/source/tailscale/client/testing.go:36]: yes, will move. some deprecated code

**Review by hermanschaaf — APPROVED** [review]

File: website/pages/docs/plugins/sources/tailscale/_meta.json [hunk 8]
```diff
@@ -1,5 +1,4 @@
 {
   "overview": "Overview",
-  "configuration": "Configuration",
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/tailscale/_meta.json:3]: Can we add a redirect from `/docs/plugins/sources/tailscale/configuration` to `/docs/plugins/sources/tailscale/overview` in `vercel.json`?

**Review by erezrokah — APPROVED** [review]
Looks good, maybe only added a redirect to `vercel.json` to avoid 404

**Pull request merged**